unified_planning/test/test_anml_writer.py

```diff
@@ -0,0 +1,230 @@
+            '[ start ] is_connected(l5, l4) := false;',
+            '[ start ] is_connected(l1, l5) := false;',
+            '[ start ] is_connected(l2, l5) := false;',
+            '[ start ] is_connected(l3, l5) := false;',
+            '[ start ] is_connected(l5, l5) := false;',
+            '[ end ] is_at(l5);\n']
+        self.assertEqual(anml_problem, '\n'.join(expected_result))
+
+    def test_matchcellar(self):
+        problem = self.problems['matchcellar'].problem
+        aw = ANMLWriter(problem)
+        anml_problem = aw.get_problem()
+        expected_result = [
+            'type Match;',
+            'type Fuse;',
+            'type boolean;',
+            'fluent boolean handfree;',
+            'fluent boolean light;',
+            'fluent boolean match_used;',
+            'fluent boolean fuse_mended;',
+            'action light_match(Match m) {',
+            '   [ start, start ] (not match_used(m));',
+            '   [ start ] match_used(m) := true;',
+            '   [ start ] light := true;',
+            '   [ end ] light := false;',
+            '}',
+            'action mend_fuse(Fuse f) {',
+            '   [ start, start ] handfree;',
+            '   [ start, end ] light;',
+            '   [ start ] handfree := false;',
+            '   [ end ] fuse_mended(f) := true;',
+            '   [ end ] handfree := true;',
+            '}',
+            'instance Match m1, m2, m3;',
+            'instance Fuse f1, f2, f3;',
+            '[ start ] light := false;',
+            '[ start ] handfree := true;',
+            '[ start ] match_used(m1) := false;',
+            '[ start ] match_used(m2) := false;',
+            '[ start ] match_used(m3) := false;',
+            '[ start ] fuse_mended(f1) := false;',
+            '[ start ] fuse_mended(f2) := false;',
+            '[ start ] fuse_mended(f3) := false;',
+            '[ end ] fuse_mended(f1);',
+            '[ end ] fuse_mended(f2);',
+            '[ end ] fuse_mended(f3);\n']
+        self.assertEqual(anml_problem, '\n'.join(expected_result))
+
+    def test_robot_fluent_of_user_type(self):
+        problem = self.problems['robot_fluent_of_user_type'].problem
+        aw = ANMLWriter(problem)
+        anml_problem = aw.get_problem()
+        expected_result = [
+            'type Location;',
+            'type Robot;',
+            'fluent Location is_at;',
+            'action move(Robot robot, Location l_from, Location l_to) {',
+            '   [ start ] (is_at(robot) == l_from);',
+            '   [ start ] (not (is_at(robot) == l_to));',
+            '   [ start ] is_at(robot) := l_to;',
+            '}',
+            'instance Location l1, l2;',
+            'instance Robot r1, r2;',
+            '[ start ] is_at(r1) := l2;',
+            '[ start ] is_at(r2) := l1;',
+            '[ end ] (is_at(r1) == l1);',
+            '[ end ] (is_at(r2) == l2);\n']
+        self.assertEqual(anml_problem, '\n'.join(expected_result))
```

Comment:
```suggestion
        expected_result = """
            type Location;
            type Robot;
            fluent Location is_at;
            action move(Robot robot, Location l_from, Location l_to) {
               [ start ] (is_at(robot) == l_from);
               [ start ] (not (is_at(robot) == l_to));
               [ start ] is_at(robot) := l_to;
            }
            instance Location l1, l2;
            instance Robot r1, r2;
            [ start ] is_at(r1) := l2;
            [ start ] is_at(r2) := l1;
            [ end ] (is_at(r1) == l1);
            [ end ] (is_at(r2) == l2);"""
        self.assertEqual(anml_problem, expected_result)
```
(I left some extra spaces, but you get the idea)